File: core/src/main/java/com/larryhsiao/nyx/core/tags/WrappedTag.java
package com.larryhsiao.nyx.core.tags;

/**
 * Wrapper class of Tag
 */
public class WrappedTag implements Tag {
    private final Tag origin;

    public WrappedTag(Tag origin) {
        this.origin = origin;
    }

    @Override
    public String title() {
        return origin.title();
    }

    @Override
    public long id() {
        return origin.id();
    }

    @Override
    public int version() {
        return origin.version();
    }

    @Override
    public boolean deleted() {
        return origin.deleted();
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/tags/WrappedTag.kt
package com.larryhsiao.nyx.core.tags

/**
 * Wrapper class of Tag
 */
open class WrappedTag(private val origin: Tag) : Tag {
    override fun title(): String {
        return origin.title()
    }

    override fun id(): Long {
        return origin.id()
    }

    override fun version(): Int {
        return origin.version()
    }

    override fun deleted(): Boolean {
        return origin.deleted()
    }
}